buildutils/torque.spec.in

```diff
@@ -61,9 +61,9 @@
 %define ac_trqauthd_sock_dir %{?trqauthd_sock_dir:%{trqauthd_sock_dir}}%{!?trqauthd_sock_dir:/tmp}
 
 ### Build Requirements
-%define breq_cpuset   %{?with_cpuset:hwloc-devel}
+%define breq_cpuset   %{!?hwloc_path:%{?with_cpuset:hwloc-devel}}
 %define breq_gui      %{?with_gui:tcl-devel tk-devel tclx}
-%define breq_numa     %{?with_cpuset:hwloc-devel}
+%define breq_numa     %{!?hwloc_path:%{?with_cpuset:hwloc-devel}}
 %define breq_munge    %{?with_munge:munge-devel}
```

Comment:
> Yes and no. I understand your point, but I think you did not (entirely) answer one of my questions; AFAICT, in my scenario here, hwloc-devel will never be installed on another system than my build environment (and even not there, since I compile from source). So the torque rpm in this case effectively seems to have no dependency at all on hwloc-devel. At build time it is otherwise provided, at install time, it is no longer relevant.

That is correct; that's how `-devel` RPMs work.  They are (in general) only required for building, not installing, and are therefore typically listed as `BuildRequires:` in the spec file (as `hwloc-devel` is with TORQUE).

Some packagers put the library's main package (`hwloc` in this instance) as a `Requires:` entry in the spec file for the application's main package (`torque`), but I prefer to allow the normal RPM library dependency gathering methods to do their job, so the spec files I write don't do that.  This **does not**, however, mean that the library RPM (`hwloc`) isn't required by the built package (`torque`); it still is!

I'm not sure what question I didn't answer or where the confusion lies.  :-(  The TORQUE build process requires `hwloc-devel` to build with cpuset and/or NUMA support, and if built on a system with `hwloc-devel` installed, the built RPMs (`torque-*.rpm`) _will_ require the `hwloc` RPM to be installed in order to install them.  You're right that the `hwloc-devel` package is not required at install time, but the `hwloc` package is.

> Building a separate rpm for hwloc-devel is probably a good idea (as I said earlier), but not really necessary either I THINK (I may be very wrong here once I actually test the cpuset/cgroup features on our el6 cluster). For package infrastructures, the case is different, but they will (should?) never have a --with-hwloc-path ...

If `hwloc-devel` doesn't get pulled in, then the dependency for `hwloc` won't get generated either.  Or, you'll have a dependency on an external library that RPM doesn't know about, and when you try to go and install the TORQUE RPMs, it will break because there's no matching library in the RPM database (even if it does exist on the system).  And if one doesn't fully grok how RPM resolves dependencies (both build-time and run-time), that's very much a rabbit hole one does not wish to enter!  :-)